src/Appwrite/Platform/Modules/Databases/Http/Databases/Collections/Attributes/Action.php

```diff
@@ -0,0 +1,652 @@
+use Appwrite\Event\Event;
+use Appwrite\Extend\Exception;
+use Appwrite\Platform\Modules\Databases\Context;
+use Appwrite\Utopia\Response;
+use Appwrite\Utopia\Response as UtopiaResponse;
+use Throwable;
+use Utopia\Database\Database;
+use Utopia\Database\Document;
+use Utopia\Database\Exception\Duplicate as DuplicateException;
+use Utopia\Database\Exception\Index as IndexException;
+use Utopia\Database\Exception\Limit as LimitException;
+use Utopia\Database\Exception\Relationship as RelationshipException;
+use Utopia\Database\Exception\Structure as StructureException;
+use Utopia\Database\Exception\Truncate as TruncateException;
+use Utopia\Database\Helpers\ID;
+use Utopia\Database\Validator\Authorization;
+use Utopia\Database\Validator\Structure;
+use Utopia\Platform\Action as UtopiaAction;
+use Utopia\Swoole\Response as SwooleResponse;
+use Utopia\Validator\Range;
+
+abstract class Action extends UtopiaAction
+{
+    /**
+     * @var string|null The current context (either 'column' or 'attribute')
+     */
+    private ?string $context = Context::DATABASE_ATTRIBUTES;
+
+    /**
+     * Get the correct response model.
+     */
+    abstract protected function getResponseModel(): string|array;
+
+    /**
+     * Set the context to either `column` or `attribute`.
+     *
+     * @throws \InvalidArgumentException If the context is invalid.
+     */
+    final protected function setContext(string $context): void
+    {
+        if (!\in_array($context, [Context::DATABASE_COLUMNS, Context::DATABASE_ATTRIBUTES], true)) {
+            throw new \InvalidArgumentException("Invalid context '$context'. Use `Context::DATABASE_COLUMNS` or `Context::DATABASE_ATTRIBUTES`");
+        }
+
+        $this->context = $context;
+    }
+
+    /**
+     * Get the current context.
+     */
+    final protected function getContext(): string
+    {
+        return $this->context;
+    }
+
+    /**
+     * Returns true if current context is Collections API.
+     */
+    final protected function isCollectionsAPI(): bool
+    {
+        // columns in tables context
+        // attributes in collections context
+        return $this->getContext() === Context::DATABASE_ATTRIBUTES;
+    }
+
+    /**
+     * Get the SDK group name for the current action.
+     *
+     * Can be used for XList operations as well!
+     */
+    final protected function getSdkGroup(): string
+    {
+        return $this->isCollectionsAPI() ? 'attributes' : 'columns';
+    }
+
+    /**
+     * Get the SDK namespace for the current action.
+     */
+    final protected function getSdkNamespace(): string
+    {
+        return $this->isCollectionsAPI() ? 'collections' : 'tables';
+    }
+
+    /**
+     * Get the appropriate parent level not found exception.
+     */
+    final protected function getParentNotFoundException(): string
+    {
+        return $this->isCollectionsAPI()
+            ? Exception::COLLECTION_NOT_FOUND
+            : Exception::TABLE_NOT_FOUND;
```

Comment:
For flexibility in future, it'll be easier to make all these checks switches instead of if's

Reply:
For future cases, sure. atm, its only two so keeping it as is. Change should be easy when we decide to add more.